docs/core/howto/ssl.rst

```diff
@@ -204,10 +199,40 @@ For example,
     options = CertificateOptions(..., dhParameters=dhParams)
 
 Another part of the TLS protocol which ``CertificateOptions`` can control is the version of the TLS or SSL protocol used.
-This is often called the context's "method".
+There are two ways to configure it, either by configuring a list of protocols, or by the legacy way of setting the internal context's "method".
+By default, Twisted will configure it to use TLSv1.0 or later and disable the insecure SSLv3 protocol, but this manual control over protocols can be helpful if you need to support legacy SSLv3 systems, or you wish to restrict it down to just the strongest of the TLS versions.
+
+You can give ``CertificateOptions`` a range of TLS versions you wish it to negotiate with the ``tlsProtocols`` argument in the initializer:
+
+.. code-block:: python
+
+    from twisted.internet.ssl import (CertificateOptions, TLSVersion,
+                                      TLSVersionRange)
+    options = CertificateOptions(
+        ...,
+        tlsProtocols=TLSVersionRange(TLSVersion.TLSv1_0,
+                                     TLSVersion.TLSv1_2))
+
```

Comment:
> Is it? If it is, then you're right, this API doesn't support that. But I wasn't aware that that was the design goal.

I'm going to say it is.  I think I get to do that because I filed #6800 ;)  (That's not to say I think you can't object to that goal.)

>  I don't see any reason to allow the user to say "I want only SSLv2 and TLSv1.1, please".

This is a straw-man, I think.  SSLv2 is almost entirely unused now.  And it's pretty radically different from any of the TLS versions, anyway.  I think the "TLSv1.0 or TLSv1.2 but not TLSv1.1" (or, after time passes, any other "recent X, recent Z, but not recent Y between them") could be useful.  Perhaps TLSvZ has the best known security properties but is not yet widely supported, TLSvX has pretty good security properties, and an attack is discovered against TLSvY.  That seems like a legitimate reason to want this feature, to me.

> If the only acceptable outcome for this feature is "allow all combinations of TLS versions" then I'm actually -1 on the feature until someone convinces me that it's necessary.

I'm open to the idea that not _all_ combinations need to be supported but I don't see a simple API that allows the combinations that seem to make sense without also allowing everything else.